hack/generator/pkg/astmodel/armconversion/convert_from_arm_function_builder.go

```diff
@@ -477,22 +438,22 @@ func (builder *convertFromArmBuilder) convertComplexMapProperty(
 	rangeStatement := &ast.RangeStmt{
 		Key:   keyIdent,
 		Value: valueIdent,
-		X:     source,
+		X:     params.source,
 		Tok:   token.DEFINE,
 		Body: &ast.BlockStmt{
 			List: builder.fromArmComplexPropertyConversion(
-				valueIdent,
-				actualDestination,
-				destinationType.ValueType(),
-				elemIdent.Name,
-				append(conversionContext, destinationType),
-				handler),
+				params.withSource(valueIdent).
```

Comment:
When you're changing so many of the fields of the struct I'd find it clearer to just make a new one - the builder syntax gets a bit unwieldy (although some of that is how it looks in a diff).

Reply:
Changed to use the constructor when I'm changing all (or almost all) of the parameters - kept some `With` methods for when it's just a few things that changed.